### PR TITLE
6359 network sysinfo provider

### PR DESCRIPTION
**Pull request by Dwordcito**
|Related issue|
|---|
|#6359|

## Description
This issue aims to refactor syscollector, to separate the provider logic of the module logic.
In this particular PR is included the implementation of BSD and Linux to obtain the information of local networks

RTR
image

### CODE REVIEW

File: src/wazuh_modules/syscollector/network/networkLinuxWrapper.h [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+        }
+        return retVal.get();
+    }
+ 
+    static std::string getRedHatDHCPStatus(const std::vector<std::string>& fields)
+    {
+        std::string retVal { "unknown" };
+        const auto value { fields.at(RHInterfaceConfig::Value) };
+        
+        if (value.compare("static") == 0 || value.compare("none") == 0 || value.compare("no") == 0)
```
Comment (konangc87) [inline src/wazuh_modules/syscollector/network/networkLinuxWrapper.h:93]: if all of these are mutually exclusive you can use a map.

File: src/wazuh_modules/syscollector/network/networkLinuxWrapper.h [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+        return retVal;
+    }
+
+    static std::string getDebianDHCPStatus(const std::string& family, const std::vector<std::string>& fields)
+    {
+        std::string retVal { "enabled" };
+        if (fields.at(DebianInterfaceConfig::Family).compare(family) == 0)
+        {
+            const auto method { fields.at(DebianInterfaceConfig::Method) };
+            if (method.compare("static") == 0 || method.compare("manual") == 0)
```
Comment (konangc87) [inline src/wazuh_modules/syscollector/network/networkLinuxWrapper.h:115]: ditto

File: src/wazuh_modules/syscollector/network/networkLinuxWrapper.h [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+            for (auto& line : lines)
+            {
+                line = Utils::rightTrim(line);
+                Utils::replaceAll(line, "\t", " ");
+                Utils::replaceAll(line, "  ", " ");
+                const auto fields { Utils::split(line, ' ') };
+
+                if (GatewayFileFields::Size == fields.size())
+                {
+                    if (fields.at(GatewayFileFields::Iface).compare(ifName) == 0)
```
Comment (konangc87) [inline src/wazuh_modules/syscollector/network/networkLinuxWrapper.h:204]: you can merge both ifs in one cond1 && cond2 statement to reduce code nesting.

File: src/wazuh_modules/syscollector/network/networkLinuxWrapper.h [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+                    {
+                        const auto address { static_cast<uint32_t>(std::stoi(fields.at(GatewayFileFields::Gateway), 0, 16)) };
+                        if (address)
+                        {
+                            retVal = std::string(inet_ntoa({ address })) + "|" + fields.at(GatewayFileFields::Metric);
```
Comment (konangc87) [inline src/wazuh_modules/syscollector/network/networkLinuxWrapper.h:209]: shouldn't you break the for when the required field is found?

File: src/wazuh_modules/syscollector/network/networkLinuxWrapper.h [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+            {
+                const auto fields { Utils::split(line, ' ') };
+                if (DebianInterfaceConfig::Size == fields.size())
+                {
+                    if (fields.at(DebianInterfaceConfig::Type).compare("iface") == 0 &&
+                        fields.at(DebianInterfaceConfig::Name).compare(ifName) == 0)
+                    {
+                        if (AF_INET == family)
+                        {
+                            retVal = getDebianDHCPStatus("inet", fields);
```
Comment (konangc87) [inline src/wazuh_modules/syscollector/network/networkLinuxWrapper.h:237]: ditto

File: src/wazuh_modules/syscollector/network/networkLinuxWrapper.h [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+                for (const auto& line : lines)
+                {
+                    const auto fields { Utils::split(line, '=') };
+                    if (fields.size() == RHInterfaceConfig::Size)
+                    {
+                        if (AF_INET == family)
+                        {
+                            if (fields.at(RHInterfaceConfig::Key).compare("BOOTPROTO") == 0)
+                            {
+                                retVal = getRedHatDHCPStatus(fields);
```
Comment (konangc87) [inline src/wazuh_modules/syscollector/network/networkLinuxWrapper.h:265]: ditto

**Review by konangc87 — CHANGES_REQUESTED** [review]

**Review by tinachoB — CHANGES_REQUESTED** [review]
Great work!

File: src/shared_modules/utils/networkUnixHelper.h [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        static void getNetworks(std::unique_ptr<ifaddrs, IfAddressSmartDeleter>& interfacesAddress, std::map<std::string, std::vector<ifaddrs*>>& networkInterfaces)
+        {
+            struct ifaddrs *ifaddr { nullptr };
+            const auto ret{getifaddrs(&ifaddr)};
+            if (ret != -1)
+            {
+                interfacesAddress.reset(ifaddr);
+                for (auto ifa = ifaddr; ifa; ifa = ifa->ifa_next)
+                {
+                    if (!(ifa->ifa_flags & IFF_LOOPBACK))
```
[the remark this answers is not in the document]
Comment (Dwordcito) [reply src/shared_modules/utils/networkUnixHelper.h:47]: this defensive check exist in the for condition!

[the remark this answers is not in the document]
Comment (tinachoB) [reply src/shared_modules/utils/networkUnixHelper.h:47]: You're right! my bad! :+1:

**Review by tinachoB — APPROVED** [review]
Amazing work!

**Review by konangc87 — APPROVED** [review]
Osom!

**Pull request merged**